fobi_custom/plugins/form_handlers/collect_data/fobi_form_handlers.py

```diff
@@ -0,0 +1,81 @@
+import json
+import datetime
+from django.utils.timezone import get_current_timezone
+from django.utils.translation import ugettext_lazy as _
+
+from fobi.base import FormHandlerPlugin, form_handler_plugin_registry
+from fobi.models import FormEntry, FormElementEntry
+
+from pldp.models import Study, Location, Survey, SurveyRow, SurveyComponent
+
+class CollectDataPlugin(FormHandlerPlugin):
+    """Just Spaces data handler plugin."""
+
+    uid = "collect_data"
+    name = _("Collect data")
+
+    def run(self, form_entry, request, form, form_element_entries=None):
+        """To be executed by handler."""
+
+        form_id = form_entry.id
+
+        # These are placeholders. Location and Study are required fields on all
+        # surveys, and we need to decide if they'll be form elements or
+        # part of the form entry itself. See https://github.com/datamade/just-spaces/issues/63
+        # Similarly, total should be a required field in observational surveys, and
+        # default to 1 for intercept surveys.
+        location = Location.objects.first()
+        study = Study.objects.first()
+
+        total = 5
+
+        # Here we are using PLDP's survey end time value to record time
+        # of submission
+        time_stop = datetime.datetime.now(tz=get_current_timezone())
+
+        new_survey = Survey.objects.create(
+            study=study,
+            form_id=form_id,
+            location=location,
+            time_stop=time_stop
+        )
+
+        print("Survey submitted " + str(time_stop))
+
+        new_survey_row = SurveyRow.objects.create(
```

Comment:
Should the creation of a SurveyRow be nested under the iteration over `form_elements` (lone 53 ff.)? If I understand correctly, we'll create a survey row (with the number people counted by the question) with each survey component? 

I also wonder if the `total` default for intercept surveys should be 1. Can you say more about your reasoning?

Reply:
Hm, excellent question! The separate SurveyRow and SurveyComponent models come from Eric's original implementation of `django-pldp`, and I've been following that: the assumption is that each submitted survey (not each question in it) will have a required field with the total number of people being counted in that survey. If it's an intercept survey, with just one respondent, the total number of people counted would be 1. This will be useful in later analysis, when we want to make claims like "x% of people were riding bicycles."

Now that we have more example PLDP documents in Gehl's repo, I can check those assumptions! It seems this understanding is supported in their suggested [data structure](https://github.com/gehl-institute/pldp/blob/master/Public%20Life%20Data%20Protocol/PLDP_1.0%20Data%20Sheet%20Template%20(Single%20Table)). 

However, not all of their sample data (like [this one](https://github.com/gehl-institute/pldp/blob/master/Sample%20Data/sf-actscan.csv)) includes that total field at all. Maybe the total default should be None, and it should be assigned to 1 for intercept surveys specifically?

Reply:
> The expectation is that the total number of people counted in observational surveys does not change from question-to-question.

Yes!

A reason to save 1 as a `total` for intercept surveys is that this would make it possible to compare across multiple submissions of the same survey form. For example, noting that out of 50 surveys, 40 of them responded "yes" to "Are you riding a bicycle?"

This is a question of how we're handling analysis, and it could also be derived later from noting that the survey is intercept. However, it makes more sense to me to save that in the `total` field while we're submitting the form and make a distinction between a survey regarding a single person and a survey regarding an unknown number of people.

Reply:
@reginafcompton what do you think about this?

Reply:
Okay, I think 1 as default for intercept surveys makes sense, given your example above. Let's plan for that. And, as you note, this particular detail may change depending on how the tool handles analysis.